### PR TITLE
jormun: Round second of datetime before calling realtime proxy to improve cachability

### PR DESCRIPTION
**Pull request by kinnou02**
The cache key used with realtime proxy contains the url called, so it
does contains the datetime with the seconds, it makes cache only working
for one second, that's not very good...
This PR round the second to a multiple of 30, giving us 30 seconds of
cache, this is also configurable with the `step` parameter.

We may want to rename this params as it's not very clear what it does
based on it's name, so if you have an idea I take it.

### CODE REVIEW

**Review by mbounabi — APPROVED** [review]

**Review by nonifier — APPROVED** [review]

File: source/jormungandr/jormungandr/realtime_schedule/timeo.py [hunk 1]
```diff
@@ -217,9 +218,9 @@ def _make_url(self, route_point, count=None, from_dt=None):
         # timeo can only handle items_per_schedule if it's < 5
         count = min(count or 5, 5)# if no value defined we ask for 5 passages
 
-        # if a custom datetime is provided we give it to timeo
+        # if a custom datetime is provided we give it to timeo but we round it to improve cachability
         dt_param = '&NextStopReferenceTime={dt}'\
-            .format(dt=self._timestamp_to_date(from_dt).strftime('%Y-%m-%dT%H:%M:%S')) \
+            .format(dt=floor_datetime(self._timestamp_to_date(from_dt), self.step).strftime('%Y-%m-%dT%H:%M:%S')) \
```
Comment (pbougue) [inline source/jormungandr/jormungandr/realtime_schedule/timeo.py]: Then we should also filter afterward, removing those in interval `[floor(from_dt), from_dt[` no?

File: source/jormungandr/jormungandr/realtime_schedule/timeo.py [hunk 2]
```diff
@@ -217,9 +218,9 @@ def _make_url(self, route_point, count=None, from_dt=None):
...
             if from_dt else ''
```
Comment (pbougue) [inline source/jormungandr/jormungandr/realtime_schedule/timeo.py]: So we don't have to modify this as it is cached on url (and it never happens)... But shouldn't we enforce that "step" and cache-duration match?

File: source/jormungandr/jormungandr/realtime_schedule/timeo.py [hunk 3]
```diff
@@ -73,6 +73,7 @@ def __init__(self, id, service_url, service_args, timezone,
         fail_max = kwargs.get('circuit_breaker_max_fail', app.config['CIRCUIT_BREAKER_MAX_TIMEO_FAIL'])
         reset_timeout = kwargs.get('circuit_breaker_reset_timeout', app.config['CIRCUIT_BREAKER_TIMEO_TIMEOUT_S'])
         self.breaker = pybreaker.CircuitBreaker(fail_max=fail_max, reset_timeout=reset_timeout)
+        self.step = kwargs.get('step', 30)
```
Comment (pbougue) [inline source/jormungandr/jormungandr/realtime_schedule/timeo.py]: same here, a comment would be nice.

File: source/jormungandr/jormungandr/realtime_schedule/siri.py [hunk 4]
```diff
@@ -108,6 +108,7 @@ def __init__(self, id, service_url, requestor_ref,
         self.instance = instance
         self.breaker = pybreaker.CircuitBreaker(fail_max=app.config.get('CIRCUIT_BREAKER_MAX_SIRI_FAIL', 5),
                                                 reset_timeout=app.config.get('CIRCUIT_BREAKER_SIRI_TIMEOUT_S', 60))
+        self.step = kwargs.get('step', 30)
```
Comment (pbougue) [inline source/jormungandr/jormungandr/realtime_schedule/siri.py]: I'd name it `from_datetime_step` but it's still not super sexy...
at least a comment is welcome: "A step is applied on from_datetime to discretize calls and allow caching them"

File: source/jormungandr/jormungandr/realtime_schedule/timeo.py (hunk 1 above)
Comment (kinnou02) [reply source/jormungandr/jormungandr/realtime_schedule/timeo.py]: We could, but we didn't do it before when we were thinking that the cache was working

File: source/jormungandr/jormungandr/realtime_schedule/timeo.py (hunk 2 above)
Comment (kinnou02) [reply source/jormungandr/jormungandr/realtime_schedule/timeo.py]: true, I will initialize `step` with `app.config['CACHE_CONFIGURATION'].get('TIMEOUT_TIMEO', 60)`

**Review by pbougue — APPROVED** [review]
Still not super fan of returning something before that param... I don't see any way an integration would fail on this, though, and as it's a small duration, it's probably fine.

**Pull request merged**